### DISCUSSION TITLE
Did firmware 4.36.10.2864 (break/stop/prevent) most future development of wz_mini_hacks

### DISCUSSION THREAD

**Discussion by capncybo** [gtxaspec/wz_mini_hacks#570] [General]
@gtxaspec (Hope you are doing okay?)
I've noticed "most" of the  wz_mini_hacks development seems to have stagnated.
Is that because when WYZE released firmware 4.36.10.2864 (November 2, 2022)...
It's essentially broken the primary (RTSP) streaming functionality, moving forward?
I've noticed it seems to be a bit more work to roll-back firmware _(See in BOLD below_)
But I'm somewhat perplexed by their current status regarding their two notes afterwards.
Regardless I've set all my camera configurations NOT to upgrade.
But I'm obviously not sure how PUSHY/Aggressive WYZE will get with their future updates.
Perhaps you are just super busy elsewhere with life & I hope you are well, Take care.

WYZE Cam v3 Firmware:
4.36.10.2864 (November 2, 2022)

    Improved Internet connection code
    Improved live stream connectivity
    Improved IoT connectivity
    Fixed an issue that prevented the device from reconnecting after a router reboot
    Fixed an issue that caused devices to stop working after a firmware update
    Fixed a bug that caused an incorrect membership status after quickly switching membership
    Fixed a bug that caused Error 3001 when viewing motion and sound Events
    Security improvements
    **Due to the security improvement on this firmware, you will need to go through the factory reset process for your camera if you go back to firmware 4.36.9.139 or below**

Note: This firmware is being paused to investigate an issue with event recording.

Note: We are doing a gradual release for this firmware over two weeks. If you do not have this firmware version yet, don't worry! You'll receive it later.

**Comment by sideup66** [comment]
Actually yeah, as it sits, the new firmware update effectively breaks ANY RTSP functionality we had. Im suspecting the way forward is possibly a patched version of the icamera bin file, maybe retrofitting it with an older version for usability of rtsp functions.... However im not sure if some of the other proposed fixes exist. Some of the network changes could be handy to possibly deprecate some fixes we have thought up over time. I personally have not upgraded as I heard many complaints about the new versions having some problems with capture and other functions. If we can get around the blockade currently in place (we still have root on the plain v3s, jury is out on the v3pro), you'll probably see development come back. Thing is, I think alot of people are holding off on this update due to the problems, and maybe waiting for the following one, which may be a big bugfix update and smooth things out.

**Comment by virmaior** [comment]
I'm actually still back on 4.36.2.5. 
I've got a pull request in for some web improvements but gtxaspec hasn't moved on it.
In general, the current system is working for me and my use case ... but I've got one camera (the one I usually test new things on) which lost all of its /configs directory at some point after I upgraded to the Oct 13 wz_mini code.  It also apparently just happened last night as well ... So my first task is to figure out why / how the entire /configs directory's contents are getting clobbered.

**Comment by sideup66** [comment]
Hopefully he didnt get some kind of cease and desist from wyze to stop development. Though im not sure how that would hold in court for them, the components modified in wz_mini_hacks are linux mainly...which is open source. Show that to any judge with the license agreements and basically theyll tell wyze to pound sand...

As for my setup, its been working fine. I dont rely on wz_mini for rtsp. The newest firmware doesnt affect anything im using, but im going to wait for a bugfix release or two before jumping. Could be that he isnt developing either as what he has/needs is working fine...

**Comment by endertable** [comment]
Yep , the current release is pretty stable so no real reason to keep doing nuance updates that can possibly break something else.  Take vimaior's issue above concerning his /configs directory.  That's challenging to troubleshoot especially if an isolated incident.  

Then there's the necessary questions for other any issues that come up: What FW are you on, what version of wyze_hacks_mini, any custom scripts running, online/offline, rtsp or no, etc, etc.

At this point I figure updates would come from:

A breakthrough on the V3 Pro
Any developer, or otherwise, that can provide hacked iCamera, or have an iCamera wrapper to work on the latest Wyze FW
Any developer, or otherwise, that can provide much requested working binaries, e.g. for simplified local web browser streaming mJpegStreamer, onvif, webrtc, etc.

Just my 2¢

**Comment by capncybo** [comment]
@virmaior re: the camera update you received last night... Yikes that is somewhat UN-nerving.
Especially considering it may have put one(or more) of these cameras into a non-functioning state.
I'd really just like to hear #1) that @gtxaspec is actually still alive & OK.
But also #2) if this project has been shutdown via C&D or if @gtxaspec has simply decided to cease future development.
I've noticed the Pull Requests just  sitting.
But in days gone bye... @gtxaspec would have likely added them soon after their submission.
Obviously his spare-time could already be maxed out elsewhere.
And if he's done with the project, it's his choice.
I just hope he's Healthy.

**Comment by skilau** [comment]
@gtxaspec has responded to threads here and there, but yeah, no idea if he is taking a break, or is working through the possible ways around Wyze's changes.

I know for myself, I am sticking pat for now.

3 of my cameras are all mounted outside, it is winter, and I live in Minnesota.
I have no plans to change anything with them for awhile, as there is no way I want to be climbing around on a ladder in our ice and snow!
(PS: If anyone wanted to know, I had them out there last winter too, and they work perfectly fine in -20F weather!)

**Comment by DAMAGEDatheCORE** [comment]
Sticking pat? 🤔

Tbh, if I were gtxaspec, I'd take a well-deserved break before diving into
trying to find workarounds for the latest firmwares.

**Comment by gtxaspec** [comment]
Howdy all,

I'm still here, really busy in real life, still developing and investigating the new releases. Working on the floodlight cam at the moment.

Also trying to get a hold of the V3 Pro, but they are sold out everywhere.

**Comment by dohlin** [comment]
> Howdy all,
> 
> I'm still here, really busy in real life, still developing and investigating the new releases. Working on the floodlight cam at the moment.
> 
> Also trying to get a hold of the V3 Pro, but they are sold out everywhere.

Great to hear!  I'm brand new to this repo, but after reading the recent concerns I too was worried.  Just wanted to say thank you so much for doing all of this; myself and many others undoubtedly so appreciate your hard work!

**Comment by MomosX** [comment]
> Howdy all,
> 
> I'm still here, really busy in real life, still developing and investigating the new releases. Working on the floodlight cam at the moment.
> 
> Also trying to get a hold of the V3 Pro, but they are sold out everywhere.

Glad , very glad you are still with us :)  You made many of us dependent on you as we love this mod. For me the ETH connection is the best feature ever !!!!

**Comment by digitalpixies** [comment]
> > Howdy all,
> > I'm still here, really busy in real life, still developing and investigating the new releases. Working on the floodlight cam at the moment.
> > Also trying to get a hold of the V3 Pro, but they are sold out everywhere.
> 
> Glad , very glad you are still with us :) You made many of us dependent on you as we love this mod. For me the ETH connection is the best feature ever !!!!

Same, I rely on the USB gadget support to get the Wyze v3 connected via USB to my rpi. This frees up wifi bandwidth and spread out the frigate nvr workload

**Comment by gtxaspec** [comment]
I think for now streaming video will survive on .139.  

Most of the features in the newer firmware are for folks who want the camera to do everything, with a simple app interface.

Personally RTSP + Home Assistant or Frigate or something else is way better.

**Comment by mitchross** [comment]
How can I downgrade the firmware if the camera already has too high of a firmware?

**Reply by sideup66** [reply]
You have to download an older version. Which camera do you have?

**Reply by mitchross** [reply]
wyze v3

**Reply by jeffmccune** [reply]
https://download.wyzecam.com/firmware/v3/demo_wcv3_4.36.9.139.zip

https://support.wyze.com/hc/en-us/articles/360031490871-How-to-flash-your-Wyze-Cam-firmware-manually

Flash Firmware for Wyze Cam v3

1. Wyze Cam v3 files can be found on the [Release Notes](https://support.wyzecam.com/hc/en-us/articles/360024852172) page.
2. Download a specific version of firmware, For Windows 10: Right-click on the folder and select extract all. For Mac: Just drag and drop the folder, put it on the root directory of your microSD card. For both Mac and Windows, rename the file to demo_wcv3.bin.
3. Power off your camera and insert your microSD card.
4. Hold the setup button, plug in your USB cable, keep holding the setup button for 3-6 seconds until the light is solid purple, then release the button.
  - Note: For "Solid Purple" both the blue and red LEDs will be on at once, making it appear to be a solid purple.
5. Wait for 3-4 minutes. The camera will reboot and change light status during this time.
6. Once this is done, your camera should be functioning with the original setup. No need to re-setup your camera!

**Reply by crossan007** [reply]
Is there any mechanism from a root ssh shell to update the firmware remotely?

I see a couple of the scripts make calls to `flashcp`, but I have not been able to find any guides that explain how to write a `demo_wcv3.bin` file to the flash of a running camera over SSH.

Anyone know if this is possible?   (specifically rolling back firmware versions without physical access)

**Comment by shandshellin** [comment]
This is a bit late. i have several v3 cams, some old some new with newer firmware so i cant downgrade them fully. It applies the old firmware and the mini_hacks does its squash bit and everything but then just sits there with a flashing red led and no dchp for my lan adapter. I made a changed to the S03ethernet.bash file and enabled dhcp with a delay after the adapter drive loaded. Now it boots and gets an ip and i can access it fine. This worked for 3 cameras that would not mini_hacks properly regardless of the sd card or fw version.



#!/bin/sh

### BEGIN INIT INFO

# Provides:
# Short-Description: USB Ethernet Support
# Description:       Enable USB Ethernet adapter support and kernel module

### END INIT INFO

. /opt/wz_mini/etc/rc.common
. /opt/wz_mini/wz_mini.conf

case "$1" in
start)

	echo "#####$(basename "$0")#####"

	if [[ "$ENABLE_USB_ETH" == "true" ]]; then

		insmod $KMOD_PATH/kernel/drivers/net/usb/usbnet.ko

		echo "Auto-Detect an Ethernet Driver and load it"
		if [[ "$ENABLE_USB_ETH_MODULE_AUTODETECT" == "true" ]]; then
			for DEVICE in `lsusb | awk '{print $6}' | tr '[:upper:]' '[:lower:]'`; do
				case $DEVICE in
					'077b:2226' | '0846:1040' | '2001:1a00' | '0b95:1720' | '07b8:420a' |\
					'08dd:90ff' | '0557:2009' | '0411:003d' | '0411:006e' | '6189:182d' |\
					'0df6:0056' | '07aa:0017' | '1189:0893' | '1631:6200' | '04f1:3008' |\
					'17ef:7203' | '0b95:772b' | '0b95:7720' | '0b95:1780' | '0789:0160' |\
					'13b1:0018' | '1557:7720' | '07d1:3c05' | '2001:3c05' | '2001:1a02' |\
					'1737:0039' | '04bb:0930' | '050d:5055' | '05ac:1402' | '0b95:772a' |\
					'14ea:ab11' | '0db0:a877' | '0b95:7e2b' | '0b95:172a' | '066b:20f9')
						echo "Loading ASIX Ethernet driver..."
						modprobe asix
						;;
					'0b95:1790' | '0b95:178a' | '0df6:0072')
						echo "Loading AX88179 Gigabit Ethernet driver..."
						modprobe ax88179_178a
						;;
					'1004:61aa' | '046d:c11f' | '1410:b001' | '1410:9010' | '413c:8195' |\
					'413c:8196' | '413c:819b' | '16d5:650a' | '12d1:14ac')
						echo "Loading USB CDC Ethernet driver..."
						modprobe cdc_ether
						;;
					'0bda:8152')
						echo "Loading Realtek RTL8152 driver..."
						modprobe r8152
						;;
				esac
			done
		fi

		echo "Manually load any other Ethernet Drivers if asked for"
		if [[ "$ENABLE_USB_ETH_MODULE_MANUAL" != "" ]]; then
			for i in $(echo "$ENABLE_USB_ETH_MODULE_MANUAL" | tr "," "\n")
			do
				insmod $KMOD_PATH/kernel/drivers/net/usb/$i.ko
			done
		fi

		# Wait for eth0 to appear
		echo "Waiting for USB Ethernet interface..."

		for i in 1 2 3 4 5 6 7 8 9 10
		do
			if [ -e /sys/class/net/eth0 ]; then
				echo "eth0 detected"
				break
			fi
			sleep 1
		done

		if [ ! -e /sys/class/net/eth0 ]; then
			echo "ERROR: eth0 was not detected"
			exit 1
		fi

		# Save Ethernet MAC address
		cat /sys/class/net/eth0/address | tr '[:lower:]' '[:upper:]' > /opt/wz_mini/tmp/eth0_mac

		# Override Ethernet MAC address if configured
		if [ -n "$USB_ETH_MAC_ADDR" ]; then
			echo "Setting USB Ethernet MAC address to $USB_ETH_MAC_ADDR"
			/opt/wz_mini/tmp/.bin/ifconfig eth0 hw ether $USB_ETH_MAC_ADDR
		fi

		# Bring interface up
		echo "Bringing eth0 up..."
		/opt/wz_mini/tmp/.bin/ifconfig eth0 up

		# Start DHCP
		echo "Starting DHCP on eth0..."

		udhcpc -i eth0 -q -n -t 10 -T 3

		if [ $? -eq 0 ]; then
			echo "DHCP successfully configured eth0"
		else
			echo "WARNING: DHCP failed on eth0"
		fi

	else
		echo "USB Ethernet disabled"
	fi

	;;
*)
	echo "Usage: $0 {start}"
	exit 1
	;;
esac
